resolwe/flow/views.py

```diff
@@ -500,6 +501,97 @@ def remove_data(self, request, pk=None):
         return Response()
 
 
+class EntityViewSet(CollectionViewSet):
+    """API view for entities."""
+
+    filter_class = EntityFilter
+    serializer_class = EntitySerializer
+
+    queryset = Entity.objects.annotate(
+        latest_date=Max('data__modified')
+    ).prefetch_related(
+        'descriptor_schema',
+        'contributor'
+    ).order_by('-latest_date')
```

Comment:
Maybe this will be too slow and we should have a _latest_date_ field that we update through signals each time a data object changes.